src/coreclr/jit/morph.cpp

```diff
@@ -17766,11 +17769,18 @@ void Compiler::fgExpandQmarkForCastInstOf(BasicBlock* block, Statement* stmt)
     cond1Block->bbJumpDest = remainderBlock;
     cond2Block->bbJumpDest = remainderBlock;
 
-    // Set the weights; some are guesses.
+    // Currently, we don't instrument internal blocks, so the only way we can set weights to these blocks
+    // is to analyze successors and take a guess.
+    float cond2BlockWeight = 0;
+    if (currBbWeight > 0)
```

Comment:
This needs a comment explaining why the computation makes sense.

I don't think values of 100 and 50 should enter in directly (the scaling should be based entirely on the weights from neighboring blocks and perhaps some ad-hoc guesses which you should call out explicitly).

Reply:
I've added some comments around (covers 50 value)